components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt

```diff
@@ -44,6 +46,26 @@ open class PlacesHistoryStorage(
     context: Context,
     crashReporter: CrashReporting? = null
 ) : PlacesStorage(context, crashReporter), HistoryStorage, HistoryMetadataStorage, SyncableStore {
+    /**
+     * Backing property for the reader used in all queries in this this class.
+     * Allows to easily change the reader in tests. Otherwise can be ignored.
+     * This allows for better encapsulation. It is `private` while [reader] is `internal`.
+     */
+    @VisibleForTesting
+    @SuppressWarnings("VariableNaming")
+    internal var _reader: PlacesReaderConnection? = null
+
+    /**
+     * Allows to read/write history data without any impact on other storage queries from other features.
+     */
+    override val reader: PlacesReaderConnection
+        // Allow the reader to be injected from tests otherwise use a quick unsafe singleton in this class.
+        get() = if (_reader != null) {
+            _reader!!
+        } else {
+            _reader = places.newReader()
+            _reader!!
+        }
```

Comment:
So, with this `PlacesHistoryStorage` has its own separate single reader (see below for a related concern). 

I was about to write that we will need to override `cleanup` method and close this reader, but remembered that we're not actually using the `cleanup` method anywhere. Reason is that it won't work, as designed, and likely doesn't need to: https://github.com/mozilla-mobile/android-components/issues/7348 -> Good reminder to deprecate this method.

From an API perspective in `PlacesHistoryStorage`:
- places().reader() will return the default reader
- reader will return the override and special reader of this sub class

Let's discuss below if a single reader is sufficient.

Reply:
Went under the impression that there is only one `places` instance with a single cached reader.
Looked into it again and saw the `PlacesStorage` only being extended by `PlacesHistoryStorage` and `PlacesBookmarksStorage`, so only they use _different_ `places` instance and different readers.
So looks to me the default reader is enough.